ERCS/erc-7811.md

```diff
@@ -0,0 +1,239 @@
+type Asset = {
+  address: Hex | "native";
+  balance: Hex;
+  type: string;
+  metadata: any;
+};
+type WalletGetAssetsResponse = Record<Hex, Asset[]>;
+```
+
+The key **SHOULD** be [EIP-155](./eip-155.md) chainId
+
+Asset fields:
+
+`address` is the address of the asset as `Hex` or `native` string for native assets.
+
+`balance` is the balance of the asset as `Hex`
+
+**`type`:** A string indicating the type of the asset. Common asset types include but **aren’t limited to**:
+
+- **`ERC20`:** For [ERC-20](./eip-20.md) tokens
+- **`ERC721`:** For [ERC-721](./eip-721.md) tokens (NFTs)
+- **`native`:** For the chain's native asset
+
+**`metadata`:** An **OPTIONAL** object containing additional information about the asset. The specific fields within the metadata object may vary depending on the asset type and the wallet's implementation.
+
+#### Example response
+
+```json
+{
+  "0x1": [
+    {
+      "address": "0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48",
+      "balance": "0xcaaea35047fe5702",
+      "type": "ERC20",
+      "metadata": {
+        "name": "Token",
+        "symbol": "TOK",
+        "decimals": 18
+      }
+    },
+    {
+      "address": "native",
+      "balance": "0xcaaea35047fe5702",
+      "type": "native"
+    }
+  ],
+  "0xa": [
+    {
+      "address": "0x456",
+      "balance": "0xcd5595",
+      "type": "ERC721",
+      "metadata": {
+        //...
+      }
+    }
+  ]
+}
+```
+
+### Asset definitions
+
+This section specifies a structure for the most commonly used asset types.
+This ERC does not specify an exhaustive list of asset types.
+Since the type is a generic string, there could be a mismatch between the type Dapp expects and the one returned by the wallet.
+It’s important that no two assets share the same type.
+Therefore, new asset types should be specified in future ERCs.
+
+**Native**
+
+```ts
+type NativeAsset = {
+  address: "native";
+  balance: Hex;
+  type: "native";
+};
+```
+
+Example:
+
+```json
+{
+  "address": "native",
+  "balance": "0xcaaea35047fe5702",
+  "type": "native"
+}
+```
+
+**ERC-20 Token**
+
+```ts
+type Erc20Asset = {
+  address: Hex;
+  balance: Hex;
+  type: "ERC20";
+  metadata: {
+    name: string;
+    symbol: string;
+    decimals: number;
+  };
+};
+```
+
+Example:
+
+```json
+{
+  "address": "0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48",
+  "balance": "0xcaaea35047fe5702",
+  "type": "ERC20",
+  "metadata": {
+    "name": "Token",
+    "symbol": "TOK",
+    "decimals": 18
+  }
+}
+```
+
+**ERC-721 Token**
+
+```ts
+type Erc721Asset = {
+  address: Hex;
+  balance: Hex;
+  type: "ERC20";
+  metadata: {
+    name: string;
+    symbol: string;
+  };
+};
+```
+
+### Capabilities
+
+If the wallet is using [CAIP-25](https://github.com/ChainAgnostic/CAIPs/blob/0848f06f6cfc29ce619bccdd5035c1d500033b21/CAIPs/caip-25.md) authorization, wallet **SHOULD** include `assetDiscovery` key in the CAIP-25 `sessionsProperties` object. Value should be an object with `supported` key and value `true`
```

Comment:
```suggestion
If the wallet is using [CAIP-25](https://github.com/ChainAgnostic/CAIPs/blob/main/CAIPs/caip-25.md) authorization, wallet **SHOULD** include `assetDiscovery` key in the CAIP-25 `sessionsProperties` object. Value should be an object with `supported` key and value `true`
```

Reply:
https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1.md#chain-agnostic-improvement-proposals-caips
Because only commit-specific links are allowed by EIP-1 (I was lobbying for links of the type proposed by ignacio but the EIP editors preferred immutable referents rather than jekyll-rendered ones; i've seen similar for W3C specs in other places, FWIW)

Reply:
as discussed `assetDiscovery` no longer needed